File: ios/BFExplorer/Plaoc/bfs/ipc/IpcStreamAbort.swift
//
//  IpcStreamAbort.swift
//  BFExplorer
//
//  Created by ui08 on 2023/2/8.
//

import Foundation

struct IpcStreamAbort {
    var type: IPC_MESSAGE_TYPE = .stream_abort
    let stream_id: String
    
    init(stream_id: String) {
        self.stream_id = stream_id
    }
}

extension IpcStreamAbort: IpcMessage {}

File: next/ts/src/core/ipc/IpcStreamAbort.ts
import { IpcMessage, IPC_MESSAGE_TYPE } from "./const.ts";

export class IpcStreamAbort extends IpcMessage<IPC_MESSAGE_TYPE.STREAM_ABORT> {
  constructor(readonly stream_id: string) {
    super(IPC_MESSAGE_TYPE.STREAM_ABORT);
  }
}
